From 1726f49af8dedf8b58cf57f80e4c86e1cb81efee Mon Sep 17 00:00:00 2001
From: Mathavan Magesh <mmagesh@maxlinear.com>
Date: Tue, 29 Oct 2024 20:35:37 +0800
Subject: [PATCH] WLANRTSYS-85963 Change log level from debug to info in frame
 handle flow

Note: Changes will be reverted once the stability issue is addressed

FIXUP to WLANRTSYS-62845 MLO changes: MLO Auth Assoc, WPA3 and eapol
changes
---
 src/ap/ieee802_11.c | 46 +++++++++++++++++++++++++++++++++++++++++++--
 1 file changed, 44 insertions(+), 2 deletions(-)

diff --git a/src/ap/ieee802_11.c b/src/ap/ieee802_11.c
index 9de2f1ade..9c84288ab 100644
--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -6956,34 +6956,48 @@ int ieee802_11_mgmt(struct hostapd_data *hapd, const u8 *buf, size_t len,
 
 	switch (stype) {
 	case WLAN_FC_STYPE_AUTH:
+#ifdef CONFIG_IEEE80211BE_MXL_MLO
+		wpa_printf(MSG_INFO, "mgmt::auth");
+#else /* CONFIG_IEEE80211BE_MXL_MLO */
 		wpa_printf(MSG_DEBUG, "mgmt::auth");
+#endif /* CONFIG_IEEE80211BE_MXL_MLO */
 		handle_auth(hapd, mgmt, len, ssi_signal, 0);
 		ret = 1;
 		break;
 	case WLAN_FC_STYPE_ASSOC_REQ:
+#ifdef CONFIG_IEEE80211BE_MXL_MLO
+		wpa_printf(MSG_INFO, "mgmt::assoc_req");
+#else /* CONFIG_IEEE80211BE_MXL_MLO */
 		wpa_printf(MSG_DEBUG, "mgmt::assoc_req");
+#endif /* CONFIG_IEEE80211BE_MXL_MLO */
 		handle_assoc(hapd, mgmt, len, 0, ssi_signal);
 		ret = 1;
 		break;
 	case WLAN_FC_STYPE_REASSOC_REQ:
+#ifdef CONFIG_IEEE80211BE_MXL_MLO
+		wpa_printf(MSG_INFO, "mgmt::reassoc_req");
+#else /* CONFIG_IEEE80211BE_MXL_MLO */
 		wpa_printf(MSG_DEBUG, "mgmt::reassoc_req");
+#endif /* CONFIG_IEEE80211BE_MXL_MLO */
 		handle_assoc(hapd, mgmt, len, 1, ssi_signal);
 		ret = 1;
 		break;
 	case WLAN_FC_STYPE_DISASSOC:
-		wpa_printf(MSG_DEBUG, "mgmt::disassoc");
 #ifdef CONFIG_IEEE80211BE_MXL_MLO
+		wpa_printf(MSG_INFO, "mgmt::disassoc");
 		mxl_mlo_handle_disassoc(hapd, mgmt, len);
 #else /* CONFIG_IEEE80211BE_MXL_MLO */
+		wpa_printf(MSG_DEBUG, "mgmt::disassoc");
 		handle_disassoc(hapd, mgmt, len);
 #endif /* CONFIG_IEEE80211BE_MXL_MLO */
 		ret = 1;
 		break;
 	case WLAN_FC_STYPE_DEAUTH:
-		wpa_msg(hapd->msg_ctx, MSG_DEBUG, "mgmt::deauth");
 #ifdef CONFIG_IEEE80211BE_MXL_MLO
+		wpa_msg(hapd->msg_ctx, MSG_INFO, "mgmt::deauth");
 		mxl_mlo_handle_deauth(hapd, mgmt, len);
 #else /* CONFIG_IEEE80211BE_MXL_MLO */
+		wpa_msg(hapd->msg_ctx, MSG_DEBUG, "mgmt::deauth");
 		handle_deauth(hapd, mgmt, len);
 #endif /* CONFIG_IEEE80211BE_MXL_MLO */
 		ret = 1;
@@ -6992,7 +7006,11 @@ int ieee802_11_mgmt(struct hostapd_data *hapd, const u8 *buf, size_t len,
 #ifdef CONFIG_VENDOR_MXL
 		mxl_hostapd_action_ssi_set(hapd, ssi_signal);
 #endif /* CONFIG_VENDOR_MXL */
+#ifdef CONFIG_IEEE80211BE_MXL_MLO
+		wpa_printf(MSG_INFO, "mgmt::action");
+#else /* CONFIG_IEEE80211BE_MXL_MLO */
 		wpa_printf(MSG_DEBUG, "mgmt::action");
+#endif /* CONFIG_IEEE80211BE_MXL_MLO */
 		ret = handle_action(hapd, mgmt, len, freq);
 		break;
 	default:
@@ -7567,30 +7585,54 @@ void ieee802_11_mgmt_cb(struct hostapd_data *hapd, const u8 *buf, size_t len,
 
 	switch (stype) {
 	case WLAN_FC_STYPE_AUTH:
+#ifdef CONFIG_IEEE80211BE_MXL_MLO
+		wpa_printf(MSG_INFO, "mgmt::auth cb");
+#else /* CONFIG_IEEE80211BE_MXL_MLO */
 		wpa_printf(MSG_DEBUG, "mgmt::auth cb");
+#endif /* CONFIG_IEEE80211BE_MXL_MLO */
 		handle_auth_cb(hapd, mgmt, len, ok);
 		break;
 	case WLAN_FC_STYPE_ASSOC_RESP:
+#ifdef CONFIG_IEEE80211BE_MXL_MLO
+		wpa_printf(MSG_INFO, "mgmt::assoc_resp cb");
+#else /* CONFIG_IEEE80211BE_MXL_MLO */
 		wpa_printf(MSG_DEBUG, "mgmt::assoc_resp cb");
+#endif /* CONFIG_IEEE80211BE_MXL_MLO */
 		handle_assoc_cb(hapd, mgmt, len, 0, ok);
 		break;
 	case WLAN_FC_STYPE_REASSOC_RESP:
+#ifdef CONFIG_IEEE80211BE_MXL_MLO
+		wpa_printf(MSG_INFO, "mgmt::reassoc_resp cb");
+#else /* CONFIG_IEEE80211BE_MXL_MLO */
 		wpa_printf(MSG_DEBUG, "mgmt::reassoc_resp cb");
+#endif /* CONFIG_IEEE80211BE_MXL_MLO */
 		handle_assoc_cb(hapd, mgmt, len, 1, ok);
 		break;
 	case WLAN_FC_STYPE_PROBE_RESP:
 		wpa_printf(MSG_EXCESSIVE, "mgmt::proberesp cb ok=%d", ok);
 		break;
 	case WLAN_FC_STYPE_DEAUTH:
+#ifdef CONFIG_IEEE80211BE_MXL_MLO
+		wpa_printf(MSG_INFO, "mgmt::deauth cb");
+#else /* CONFIG_IEEE80211BE_MXL_MLO */
 		wpa_printf(MSG_DEBUG, "mgmt::deauth cb");
+#endif /* CONFIG_IEEE80211BE_MXL_MLO */
 		handle_deauth_cb(hapd, mgmt, len, ok);
 		break;
 	case WLAN_FC_STYPE_DISASSOC:
+#ifdef CONFIG_IEEE80211BE_MXL_MLO
+		wpa_printf(MSG_INFO, "mgmt::disassoc cb");
+#else /* CONFIG_IEEE80211BE_MXL_MLO */
 		wpa_printf(MSG_DEBUG, "mgmt::disassoc cb");
+#endif /* CONFIG_IEEE80211BE_MXL_MLO */
 		handle_disassoc_cb(hapd, mgmt, len, ok);
 		break;
 	case WLAN_FC_STYPE_ACTION:
+#ifdef CONFIG_IEEE80211BE_MXL_MLO
+		wpa_printf(MSG_INFO, "mgmt::action cb ok=%d", ok);
+#else /* CONFIG_IEEE80211BE_MXL_MLO */
 		wpa_printf(MSG_DEBUG, "mgmt::action cb ok=%d", ok);
+#endif /* CONFIG_IEEE80211BE_MXL_MLO */
 		handle_action_cb(hapd, mgmt, len, ok);
 		break;
 	default:
-- 
2.43.0

